Realm/Tests/RealmTests.mm

```diff
@@ -91,6 +91,40 @@ - (void)testReadOnlyFile {
     [NSFileManager.defaultManager setAttributes:@{NSFileImmutable: @NO} ofItemAtPath:RLMTestRealmPath() error:nil];
 }
 
+- (void)testReadOnlyFileInImmutableDirectory {
+    @autoreleasepool {
+        RLMRealm *realm = self.realmWithTestPath;
+        [realm beginWriteTransaction];
+        [StringObject createInRealm:realm withValue:@[@"a"]];
+        [realm commitWriteTransaction];
+    }
+
+    // Delete `*.lock` and `.note` files to simulate opening Realm in an app bundle
+    NSString *testRealmPath = RLMTestRealmPath();
+    [[NSFileManager defaultManager] removeItemAtPath:[testRealmPath stringByAppendingString:@".lock"] error:nil];
+    [[NSFileManager defaultManager] removeItemAtPath:[testRealmPath stringByAppendingString:@".note"] error:nil];
+
+    // Make parent directory immutable to simulate opening Realm in an app bundle
+    NSString *parentDirectoryOfTestRealmPath = [RLMTestRealmPath() stringByDeletingLastPathComponent];
+    [NSFileManager.defaultManager setAttributes:@{NSFileImmutable: @YES} ofItemAtPath:parentDirectoryOfTestRealmPath error:nil];
+
+    RLMRealm *realm;
+    // Read-only Realm should be opened even in immutable directory
+    XCTAssertNoThrow(realm = [self readOnlyRealmWithPath:RLMTestRealmPath() error:nil]);
+
+    XCTestExpectation *createdRealmInSubThread = [self expectationWithDescription:@"Instantiate Realm in sub thread"];
+    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
```

Comment:
This can just be `[self dispatchAsyncAndWait:^{ [self readOnlyRealmWithPath:RLMTestRealmPath() error:nil]; }]`, which also handles ensuring that the RLMRealm created on the background thread is deallocated before the test ends.